tests/parametric/test_telemetry.py

```diff
@@ -254,3 +254,86 @@ def test_telemetry_event_not_propagated(self, library_env, test_agent, test_libr
+
+
+@rfc("https://docs.google.com/document/d/1xTLC3UEGNooZS0YOYp3swMlAhtvVn1aa639TGxHHYvg/edit")
+@scenarios.parametric
+@features.telemetry_app_started_event
+class Test_TelemetrySCAEnvVar:
+    """
+    This telemetry entry has the value of DD_APPSEC_SCA_ENABLED in the library.
+    """
+
+    @pytest.mark.parametrize(
+        "library_env, outcome_value",
+        [
+            ({**DEFAULT_ENVVARS, "DD_APPSEC_SCA_ENABLED": "true",}, "true"),
+            ({**DEFAULT_ENVVARS, "DD_APPSEC_SCA_ENABLED": "True",}, "true"),
+            ({**DEFAULT_ENVVARS, "DD_APPSEC_SCA_ENABLED": "1",}, "true"),
+            ({**DEFAULT_ENVVARS, "DD_APPSEC_SCA_ENABLED": "false",}, "false"),
+            ({**DEFAULT_ENVVARS, "DD_APPSEC_SCA_ENABLED": "False",}, "false"),
+            ({**DEFAULT_ENVVARS, "DD_APPSEC_SCA_ENABLED": "0",}, "false"),
+        ],
+    )
+    def test_telemetry_sca_enabled_propagated(self, library_env, outcome_value, test_agent, test_library):
+        with test_library.start_span("first_span"):
+            pass
+
+        test_agent.wait_for_telemetry_event("app-started")
+        requests = test_agent.raw_telemetry(clear=True)
+        assert len(requests) > 0, "There should be at least one telemetry event (app-started)"
+        for req in requests:
+            body = json.loads(base64.b64decode(req["body"]))
+            if body["request_type"] != "app-started":
+                continue
+            assert (
+                "configuration" in body["payload"]
+            ), "The configuration should be included in the telemetry event, got {}".format(body)
+
+            configuration = body["payload"]["configuration"]
+
+            configuration_by_name = {item["name"]: item for item in configuration}
+
+            DD_APPSEC_SCA_ENABLED = "DD_APPSEC_SCA_ENABLED"
+            if context.library in ("dotnet", "jvm", "nodejs"):
+                DD_APPSEC_SCA_ENABLED = "appsec.sca.enabled"
+            elif context.library == "php":
+                DD_APPSEC_SCA_ENABLED = "appsec.sca_enabled"
+
+            cfg_appsec_enabled = configuration_by_name.get(DD_APPSEC_SCA_ENABLED)
+            assert cfg_appsec_enabled is not None, "Missing telemetry config item for '{}'".format(
+                DD_APPSEC_SCA_ENABLED
+            )
+
+            if context.library in ("golang", "jvm", "dotnet"):
+                outcome_value = True if outcome_value == "true" else False
+
+            assert cfg_appsec_enabled.get("value") == outcome_value
+
+    @pytest.mark.parametrize("library_env", [{**DEFAULT_ENVVARS}])
+    def test_telemetry_sca_enabled_not_propagated(self, library_env, test_agent, test_library):
+        with test_library.start_span("first_span"):
+            pass
+
+        test_agent.wait_for_telemetry_event("app-started")
+        requests = test_agent.raw_telemetry(clear=True)
+        assert len(requests) > 0, "There should be at least one telemetry event (app-started)"
+        for req in requests:
```

Comment:
Perhaps some of the duplicate logic could be eliminated by copying it to a common method.